File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.java
class Solution {
    public int findLongestChain(int[][] pairs) {
        // Sort pairs based on the second element of each pair
        // Arrays.sort(pairs, new Comparator<int[]>() {
        //     @Override
        //     public int compare(int[] a, int[] b) {
        //         return Integer.compare(a[1], b[1]);
        //     }
        // });
         Arrays.sort(pairs, (a, b) -> a[1] - b[1]);

        int count = 1;
        int prev = pairs[0][1]; // Initialize with a very small value

        for (int i = 1; i < pairs.length; i++) {
            if (prev < pairs[i][0]) {
                count++;
                prev = pairs[i][1];
            }
        }

        return count;
    }
}

File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.cpp
/*

pairs:{{-9, 8}, {-6, -2}, {-6, 9}, {-5, 3}, {-1, 4}, {0, 3}, {1, 6}, {8, 10}}
count:1
prev:8

*/

class Solution {
public:
    int findLongestChain(vector<vector<int>>& pairs) {
        // Sort the pairs by the second element of each vector
        sort(pairs.begin(), pairs.end(), [](const vector<int>& a, const vector<int>& b) {
            return a[1] < b[1];
            //return b[1]-a[1];
        });

        int count = 0;
        int prev = INT_MIN; // Initialize with a very small value

        for (const auto& p : pairs) {
            if (prev < p[0]) {
                count++;
                prev = p[1];
            }
        }

        return count;
    }
};